src/main/lighting/LightSource.java

```diff
@@ -0,0 +1,33 @@
+package lighting;
+
+import primitives.Colour;
+import primitives.NormalizedVector;
+import primitives.Point;
+
+/**
+ * Abstract base class for all lights which have a source.
+ *
+ * @author Abraham Murciano
+ * @author Eli Levin
+ */
+public abstract class LightSource extends Light {
+	protected LightSource(Colour colour) {
+		super(colour);
+	}
+
+	/**
+	 * Calculate the colour of the light that this light source is projecting onto a given point.
+	 *
+	 * @param point The point at which to calculate the colour of the light.
+	 * @return The colour of the light at the given point.
+	 */
+	public abstract Colour colourAt(Point point);
```

Comment:
wots this?

Reply:
imagine a light bulb. the colour of the light at a point close to the light bulb is a lot brighter than the colour at a point far away from it.

Reply:
:+1: